### PR TITLE
feat: Add a Uno-specific to access the native Window

### LINKED ISSUE #13818: [GTK, WPF] Access the native window
In Uno 4.10 I used to set the Gtk window size to not resizable, and a fixed size (to mimic my hardware) by setting GtkHost.Window.Resizable and Size, because GtkHost.Window was a static. That was changed in 5.0, so my question is how can I do this now? I tried the ApplicationView, but it doesn't do anything. What's the preferred way to set the Gtk Window size in Uno5? I've only tested this on Gtk Skia on Windows.

_Originally posted by @HakanL in https://github.com/unoplatform/uno/discussions/13806_

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #13818

## PR Type

What kind of change does this PR introduce?

- Feature

### <samp>🤖 Generated by Copilot at 40ffac7</samp>

This pull request adds the `WindowHelper.GetNativeWindow` method and the `NativeWindow` property to the `Window` class, to enable getting the native window object for each platform. It also updates the documentation and the platform-specific window classes to use these features.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/articles/features/windows-ui-xaml-window.md [hunk 1]
```diff
@@ -20,4 +20,19 @@ This feature can help blend the background of the window with the background of
...
+| `WindowHelper.SetBackground` |   ✔️     |    ✔️    | ❌    |  ❌     |  ❌  |   ❌     |  ❌         |
+
+## Retrieving the native window
+
+On some platforms, the window is backed by a native type. For those cases, you can use the `GetNativeWindow` helper:
+
+```
+#if HAS_UNO
+var nativeWindow = Uno.UI.Xaml.WindowHelper.GetNativeWindow(Window.Current);
+```
```
Comment (Youssef1313) [inline doc/articles/features/windows-ui-xaml-window.md:32]:
~~~suggestion
```csharp
#if HAS_UNO
var nativeWindow = Uno.UI.Xaml.WindowHelper.GetNativeWindow(Window.Current);
#endif
```
~~~

File: doc/articles/features/windows-ui-xaml-window.md [hunk 2]
```diff
@@ -20,4 +20,20 @@ This feature can help blend the background of the window with the background of
...
+
+On some platforms, the window is backed by a native type. For those cases, you can use the `GetNativeWindow` helper:
+
+```csharp
+#if HAS_UNO
+var nativeWindow = Uno.UI.Xaml.WindowHelper.GetNativeWindow(Window.Current);
+#endif
+```
+
+The `nativeWindow` is an object, so you need to cast it to the specific type on the given platform. See the table below:
```
Comment (ajpinedam) [inline doc/articles/features/windows-ui-xaml-window.md:35]:
```suggestion
The `nativeWindow` is an `object,` so you must cast it to the specific type on the given platform. See the table below:
```

File: doc/articles/features/windows-ui-xaml-window.md [hunk 3]
```diff
@@ -20,4 +20,20 @@ This feature can help blend the background of the window with the background of
...
+
+|                            | Skia+GTK | Skia+WPF | iOS   | Android | macOS | Catalyst | WebAssembly |
+| -------------------------- | :------: | :------: | :---: | :-----: | :---: | :------: | :---------: |
+| `WindowHelper.GetNativeWindow`` | `Gtk.Window` |  `System.Windows.Window`    | `UIKit.UIWindow`    |  `Android.App.Activity`     |  `AppKit.NSWindow`  |   `UIKit.UIWindow`     |  `null`         |
```
Comment (ajpinedam) [inline doc/articles/features/windows-ui-xaml-window.md:39]: Did not realize till now that Android does not have a `Window` type

File: doc/articles/features/windows-ui-xaml-window.md (hunk 2 above)
Comment (ajpinedam) [reply doc/articles/features/windows-ui-xaml-window.md:35]: I believe this way makes it clear that we mean "of type object"

File: src/Uno.UI/UI/Xaml/Window.Android.cs [hunk 4]
```diff
@@ -37,6 +37,8 @@ public sealed partial class Window
 				+= RaiseNativeSizeChanged;
 		}
 
+		internal Activity NativeWindow => ApplicationActivity.Instance;
```
Comment (ajpinedam) [inline src/Uno.UI/UI/Xaml/Window.Android.cs:40]: So there's nothing higher than an Activity for Android?

**Review by ajpinedam — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Window.Android.cs (hunk 4 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Window.Android.cs]: Good point! There is one - and I adjusted the code to return `Android.Views.Window` instead, which still can access the activity if needed 👍

File: src/Uno.UI/UI/Xaml/WindowHelper.cs [hunk 5]
```diff
@@ -1,22 +1,29 @@
...
+		window.Background = background;
 	}
+
+	/// <summary>
+	/// Retrieves the native Window object backing the given window.
+	/// Only applicable on some targets.
+	/// </summary>
+	/// <param name="window">Window.</param>
+	/// <returns>Native window type or null.</returns>
+	public static object GetNativeWindow(this Window window) => window.NativeWindow;
```
Comment (kazo0) [inline src/Uno.UI/UI/Xaml/WindowHelper.cs:28]: From an Uno app, how would I go about getting the `Microsoft.UI.Xaml.Window` instance from the context of a specific `UIElement` instance?

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/WindowHelper.cs:28]: I will probably be adding some Uno-specific API for this, but that will be relevant for the multi window feature. However, in many cases the `XamlRoot` will be enough

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**